packages/core/src/controllers/ConnectionController.ts

```diff
@@ -86,12 +87,46 @@ export const ConnectionController = {
   async connectWalletConnect() {
     StorageUtil.setConnectedConnector('WALLET_CONNECT')
 
-    await ChainController.state?.universalAdapter?.connectionControllerClient?.connectWalletConnect?.(
-      uri => {
-        state.wcUri = uri
-        state.wcPairingExpiry = CoreHelperUtil.getPairingExpiry()
+    if (CoreHelperUtil.isTelegram()) {
+      if (wcConnectionPromise) {
+        try {
+          await wcConnectionPromise
+        } catch (error) {
+          /* Empty */
+        }
+        wcConnectionPromise = undefined
+
+        return
+      }
+
+      if (Date.now() < (state?.wcPairingExpiry || 0)) {
```

Comment:
Let's use `CoreHelperUtil.isPairingExpired` for this to remove code duplication?